pkg/subctl/cmd/utils/restconfig/restconfig.go

```diff
@@ -0,0 +1,162 @@
+	rules.ExplicitPath = kubeConfigPath
+
+	contexts := []string{}
+	if len(kubeContexts) > 0 {
+		contexts = append(contexts, kubeContexts...)
+	} else {
+		kubeConfig := clientcmd.NewNonInteractiveDeferredLoadingClientConfig(rules, overrides)
+		rawConfig, err := kubeConfig.RawConfig()
+		if err != nil {
+			return restConfigs, err
+		}
+		for context := range rawConfig.Contexts {
+			contexts = append(contexts, context)
+		}
+	}
+
+	for _, context := range contexts {
+		if context != "" {
+			overrides.CurrentContext = context
+			config, err := GetClientConfigAndClusterName(rules, overrides)
+			if err != nil {
+				return nil, err
+			}
+
+			restConfigs = append(restConfigs, config)
+		}
+	}
+
+	return restConfigs, nil
+}
+
+func GetBrokerRestConfigAndNamespace(submariner *v1alpha1.Submariner,
+	serviceDisc *v1alpha1.ServiceDiscovery) (*rest.Config, string, error) {
+	if submariner != nil {
+		// Try to authorize against the submariner Cluster resource as we know the CRD should exist and the credentials
+		// should allow read access.
+		restConfig, _, err := resource.GetAuthorizedRestConfig(submariner.Spec.BrokerK8sApiServer, submariner.Spec.BrokerK8sApiServerToken,
+			submariner.Spec.BrokerK8sCA, rest.TLSClientConfig{}, schema.GroupVersionResource{
+				Group:    subv1.SchemeGroupVersion.Group,
+				Version:  subv1.SchemeGroupVersion.Version,
+				Resource: "clusters",
+			}, submariner.Spec.BrokerK8sRemoteNamespace)
+
+		return restConfig, submariner.Spec.BrokerK8sRemoteNamespace, err
+	}
+
+	if serviceDisc != nil {
+		// Try to authorize against the ServiceImport resource as we know the CRD should exist and the credentials
+		// should allow read access.
+		restConfig, _, err := resource.GetAuthorizedRestConfig(serviceDisc.Spec.BrokerK8sApiServer, serviceDisc.Spec.BrokerK8sApiServerToken,
+			serviceDisc.Spec.BrokerK8sCA, rest.TLSClientConfig{}, schema.GroupVersionResource{
+				Group:    "multicluster.x-k8s.io",
+				Version:  "v1alpha1",
+				Resource: "serviceimports",
+			}, serviceDisc.Spec.BrokerK8sRemoteNamespace)
+
+		return restConfig, serviceDisc.Spec.BrokerK8sRemoteNamespace, err
+	}
+
+	return nil, "", nil
+}
+
+func GetClientConfigAndClusterName(rules *clientcmd.ClientConfigLoadingRules, overrides *clientcmd.ConfigOverrides) (RestConfig, error) {
+	config := clientcmd.NewNonInteractiveDeferredLoadingClientConfig(rules, overrides)
+	clientConfig, err := config.ClientConfig()
+	if err != nil {
+		return RestConfig{}, err
+	}
+
+	raw, err := config.RawConfig()
+	if err != nil {
+		return RestConfig{}, err
+	}
+
+	clusterName := GetClusterNameFromContext(raw, overrides.CurrentContext)
+
+	if clusterName == nil {
+		return RestConfig{}, fmt.Errorf("could not obtain the cluster name from kube config: %#v", raw)
+	}
+
+	return RestConfig{Config: clientConfig, ClusterName: *clusterName}, nil
+}
+
+func GetClients(config *rest.Config) (dynamic.Interface, kubernetes.Interface, error) {
```

Comment:
```suggestion
func Clients(config *rest.Config) (dynamic.Interface, kubernetes.Interface, error) {
```

https://golang.org/doc/effective_go#Getters